src/libawkward/Index.cpp

```diff
@@ -84,7 +84,6 @@ namespace awkward {
   IndexOf<T>::IndexOf(int64_t length, kernel::lib ptr_lib)
     : ptr_(kernel::malloc<T>(ptr_lib, length * (int64_t)sizeof(T)))
     , ptr_lib_(ptr_lib)
-    , offset_(0)
```

Comment:
Why was this removed? This particular constructor should set `offset_` to `0` (because we're initializing the array right here; we know it has an offset of `0`). The others should set it to a given value.